Microsoft.Toolkit.Uwp.UI/Extensions/FrameworkElement/FrameworkElementEx.cs

```diff
@@ -0,0 +1,145 @@
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Linq;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+
+namespace Microsoft.Toolkit.Uwp.UI.Extensions
+{
+    /// <summary>
+    /// Provides attached dependency properties for the <see cref="Windows.UI.Xaml.FrameworkElement"/>
+    /// </summary>
+    public static partial class FrameworkElementEx
+    {
+        /// <summary>
+        /// Attached <see cref="DependencyProperty"/> for enabling actual size binding on any  <see cref="Windows.UI.Xaml.FrameworkElement"/>.
+        /// </summary>
+        public static readonly DependencyProperty EnableActualSizeBindingProperty = DependencyProperty.RegisterAttached("EnableActualSizeBinding", typeof(bool), typeof(FrameworkElementEx), new PropertyMetadata(null, OnEnableActualSizeBindingtPropertyChanged));
+
+        /// <summary>
+        /// Gets the <see cref="bool"/> that enables/disables actual size binding update.
+        /// </summary>
+        /// <param name="obj">The <see cref="FrameworkElement"/> to get the associated <see cref="bool"/> from</param>
+        /// <returns>The <see cref="bool"/> associated with the <see cref="FrameworkElement"/></returns>
+        public static bool GetEnableActualSizeBinding(FrameworkElement obj)
+        {
+            return (bool)obj.GetValue(EnableActualSizeBindingProperty);
+        }
+
+        /// <summary>
+        /// Sets the <see cref="bool"/> that enables/disables actual size binding update.
+        /// </summary>
+        /// <param name="obj">The <see cref="FrameworkElement"/> to associate the <see cref="bool"/> with</param>
+        /// <param name="value">The <see cref="bool"/> for binding to the <see cref="FrameworkElement"/></param>
+        public static void SetEnableActualSizeBinding(FrameworkElement obj, bool value)
+        {
+            obj.SetValue(EnableActualSizeBindingProperty, value);
+        }
+
+        private static void OnEnableActualSizeBindingtPropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+        {
+            if (!(sender is FrameworkElement baseElement))
+            {
+                return;
+            }
+
+            if ((bool) args.NewValue)
+            {
+                // Size may have changed while this was disabled, so we force an updated once user enables it
+                UpdateActualSizeProperties(baseElement, null);
+
+                // Subscribe to event
```

Comment:
Do we need to unsubscribe from the event here first just in case?

Reply:
I don't think it's needed.
Ending up on this line while being already subscribed to the event means that the propertychanged callback is called even when the value didn't change, so we should be quite safe.

I've no issues in adding it if you can provide a sample case where it's needed, but I'm quite sure we're good with this code.